src/V3Randomize.cpp

```diff
@@ -1523,9 +1675,93 @@ class RandomizeVisitor final : public VNVisitor {
             return;
         }
 
+        if (nodep->name() != "randomize") return;
+
+        if (nodep->pinsp()) {
+            // This assumes arguments to always be a member sel from nodep->fromp(), if applicable
+            // e.g. LinkDot transformed a.randomize(b, a.c) -> a.randomize(a.b, a.c)
+            // Merge pins with common prefixes so that setting their rand mode doesn't interfere
```

Comment:
The containing function is already huge - move this to another function and call it?